### PR TITLE
Fuzz rpcap

### PR DESCRIPTION
**Pull request by catenacyber**
The interesting target seems to be the server.
There is still performance optimization to be done.

This pull request also includes a small leak fix in pcap-linux.c (as well as the buffer overflows in #797)

### CODE REVIEW

**Comment by guyharris** [comment]
There is still "making it build with CMake" work to be done. It is not building with CMake on Travis on Linux or macOS, and it is not building on AppVeyor on Windows (that is only done with CMake).

**Comment by catenacyber** [comment]
@guyharris builds are now green, thanks :-)

The new fuzz targets get only built if `FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` is defined in CFLAGS, because otherwise, we do not "fake" the sockets with a predefined buffer

File: rpcapd/daemon.c [hunk 1]
```diff
@@ -2855,7 +2861,7 @@ void sleep_secs(int secs)
 {
 #ifdef _WIN32
```
Comment (guyharris) [inline rpcapd/daemon.c:2862]: It's probably cleaner to wrap the entire routine in

    #ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION

        ...

    #endif

File: sockutils.c [hunk 2]
```diff
@@ -794,7 +819,9 @@ int sock_send(SOCKET sock, SSL *ssl _U_NOSSL_, const char *buffer, size_t size,
 		if (ssl) return ssl_send(ssl, buffer, remaining, errbuf, errbuflen);
 #endif
 
-#ifdef MSG_NOSIGNAL
```
Comment (guyharris) [inline sockutils.c:797]: Same here.

File: sockutils.c [hunk 3]
```diff
@@ -1004,7 +1031,9 @@ int sock_recv(SOCKET sock, SSL *ssl _U_NOSSL_, void *buffer, size_t size,
 	 * Win32.
 	 */
 	for (;;) {
-#ifdef HAVE_OPENSSL
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+		nread = fuzz_recv(bufp, remaining);
+#elif defined(HAVE_OPENSSL)
 		if (ssl)
```
Comment (guyharris) [inline sockutils.c:1037]: You can't remove `#ifdef HAVE_OPENSSL` - this has to build even without OpenSSL.

File: rpcapd/daemon.c (hunk 1 above)
Comment (catenacyber) [reply rpcapd/daemon.c:2862]: ok

File: sockutils.c (hunk 2 above)
Comment (catenacyber) [reply sockutils.c:797]: Not sure it is relevant here, but doing it

File: sockutils.c (hunk 3 above)
Comment (catenacyber) [reply sockutils.c:1037]: It is not removed, it is moved 3 lines below.
fuzz socket now overrides both ssl and non-ssl sockets

**Pull request merged**